File: Exercice5/main.ts
// Créez une classe représentant une place de concert. Le constructeur de cette classe demande cinq paramètre dont un optionnel.
// La classe possède une méthode retournant en chaîne de caractères les détails de la place de concert sous forme de HTML.
// Exemple basique de chaîne de caractères retournée :
// "<p>Numéro de place : ${ this.username }</p>
// <p>Heure du concert : ${ this.hours }</p>"
//
// Dans un algorithme, créez deux instances de cette classe (la 1ère avec un paramètre optionnel,
// l’autre sans le paramètre optionnel), et afficher les détails de chaque objet crée.

//CLASS
class ConcertTicket {

  public concertPlace: string;
  public artistName: string;
  public dateOfConcert: string;
  public clientName: string = ""; //optionnel
  public placeNumber: number ;

//Constructeur
  constructor(concertPlace: string, artistName: string, dateOfConcert: string, placeNumber: number, clientName?: string) {
    this.concertPlace = concertPlace;
    this.artistName = artistName;
    this.dateOfConcert = dateOfConcert;
    this.placeNumber = placeNumber;
    if (clientName) {
        this.clientName = clientName;
    }

  }

//Methode
  htmlDetails(): string {
    let htmlContent: string = `
    <p>Lieu du concert: ${this.concertPlace}</p>
    <p>Nom de l'artiste: ${this.artistName}</p>
    <p>Date du concert: ${this.dateOfConcert}</p>
    <p>Numero de la place: ${this.placeNumber}</p> </br>`;

    if(this.clientName != ""){
      htmlContent = htmlContent +`<p>Nom du propréitaire: ${this.clientName}</p>`;
    };
    return (htmlContent);
}
}
let ticket01: ConcertTicket = new ConcertTicket("Opera de Paris","Mika","12/02/2020",156,"Madina" );
let ticket02: ConcertTicket = new ConcertTicket("Marseille","Soolking","16/02/2020",120);

console.log(ticket01.htmlDetails());
console.log(ticket02.htmlDetails());

File: Exercice5/main.java

class Main {
    public static void main(String[ ] args) {
    ConcertTicket ticket01 = new ConcertTicket("Opera de Paris", "Mika", "12/02/2020", 156, "Madina");
    ConcertTicket ticket02 = new ConcertTicket("Marseille", "Soolking", "16/02/2020", 120);
        System.out.println(ticket01.htmlDetails());
        System.out.println(ticket02.htmlDetails());
    }
}

//Class
class ConcertTicket {

    String concertPlace;
    String artistName;
    String dateOfConcert;
    String clientName =""; //Optionnal
    int placeNumber;


//Constructor
    public ConcertTicket(String concertPlace, String artistName, String dateOfConcert, int placeNumber ){
    this.concertPlace = concertPlace;
    this.artistName = artistName;
    this.dateOfConcert = dateOfConcert;
    this.placeNumber = placeNumber;
    }
    public ConcertTicket(String concertPlace, String artistName, String dateOfConcert, int placeNumber, String clientName){
    this.concertPlace = concertPlace;
    this.artistName = artistName;
    this.dateOfConcert = dateOfConcert;
    this.placeNumber = placeNumber;
    this.clientName = clientName;
    }
//Methode
    public String htmlDetails(){
        
        String htmlContent = "<p>Lieu du concert: "+ this.concertPlace + "</p> <p>Nom de l'artiste: "+ this.artistName + "</p> <p>Date du concert: " + this.dateOfConcert +"</p> <p>Numero de la place" +  this.placeNumber+  "</p> <p>Titulaire: "+ this.clientName + "</p>";
            
       if(this.clientName !=""){
            
          htmlContent = htmlContent + "<p> Titulaire :" + this.clientName + "</p>";  
        }
        return htmlContent;
    }

}